### PR TITLE
DEV: Remove ._buf on RandomTextIO

### LINKED ISSUE #67: Remove explicit ._buf in RandomTextIO?
Do we still need to retain the [reference to `RandomTextIO()._buf`](https://github.com/bskinn/stdio-mgr/blob/09969b383438ac349a14a56f7268e036d8548876/src/stdio_mgr/stdio_mgr.py#L85-L86), now that access to the buffer contents after close is provided by the [callback in `_PersistedBytesIO`](https://github.com/bskinn/stdio-mgr/blob/09969b383438ac349a14a56f7268e036d8548876/src/stdio_mgr/stdio_mgr.py#L62)?

### PR DESCRIPTION
**Pull request by bskinn**
No longer needed now that the callback method for persisting the
underlying stream contents is implemented.

The two test asserts for `._buf` have to just be removed, since direct access to
the underlying buffer is no longer available, and thus there is nothing
to compare the detached `f` to.

Closes #67.

### CODE REVIEW

File: tests/test_stdiomgr_base.py [hunk 1]
```diff
@@ -468,8 +468,6 @@ def test_stdout_detached(convert_newlines):
         f = o.detach()
 
         assert isinstance(f, io.BufferedRandom)
-        assert f is o._buf
```
Comment (jayvdb) [inline tests/test_stdiomgr_base.py]: these assertions are almost enough to warrant keeping a reference, as a very blunt sanity check.

but I think `o.buffer` and `i.tee.buffer` *before* the `.detach()` are the same object, and could be stored in a variable of the test method and used in these assertions after the `.detach()`.

I half expect that we will need to fiddle with the `._buf` when we tackle encoding.

Comment (bskinn) [reply tests/test_stdiomgr_base.py]: Overall, I'm +0 on removing the `._buf` reference -- I agree, it could easily turn out to be useful/necessary down the line. If we end up wanting to muck about with the underlying buffers in the real `sys.stdio` (cf #75), my hunch is we'd also face wanting to muck about with the buffers in the mocks, even if not for the same reasons.

So, in the end, I'll not merge this at this time. If we approach a 2.0 release and find that it's truly not useful, we can trim it out then.
